uber/tasks/email.py

```diff
@@ -155,33 +155,39 @@ def send_automated_emails():
             .filter(*AutomatedEmail.filters_for_active) \
             .options(joinedload(AutomatedEmail.emails)).all()
 
-        for automated_email in active_automated_emails:
-            """ This appears to be breaking our email-sending process and I do not have the time or tools to figure it out
-            automated_email.currently_sending = True
-            session.add(automated_email)
-            session.commit()"""
-            automated_email.unapproved_count = 0
-
         automated_emails_by_model = groupify(active_automated_emails, 'model')
 
         for model, query_func in AutomatedEmailFixture.queries.items():
             model_instances = query_func(session)
             for model_instance in model_instances:
                 automated_emails = automated_emails_by_model.get(model.__name__, [])
                 for automated_email in automated_emails:
+                    if automated_email.currently_sending:
+                        if automated_email.last_send_time:
+                            if (datetime.now() - automated_email.last_send_time) < timedelta(seconds=600):
```

Comment:
This is only ten minutes, but the task runs every fifteen minutes -- won't this then never be triggered?

Reply:
My assumption is that the first task has died partway through, leaving the `currently_sending=true` lock enabled, but never returning to clean it up.

This code should detect that no emails have actually been getting sent, and carry on as though `currently_sending` was false.

Reply:
Oh, also this timestamp will get updated after each email is sent (on line 182) meaning that if the last worker is still actually sending emails then it should be a recent timestamp. If the last worker died then it'll get old and this new worker will know to carry right along anyways.

One other effect of this change is that if one worker grabs the first automated email then a new worker can work on any other automated email since I moved the locking into the middle of the loop. Not sure if that's a good idea or not.

Reply:
Oh whoops, I got the idea of the code backwards. And yea, doing it per automated email category sounds fine.

I don't have a good way of testing this comparable to our live server. We only have one more big mass email to send, does merging this to next year sound okay?

Reply:
Next year is fine with me!